src/transformStateWithClones.js

```diff
@@ -8,6 +8,48 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const results = [];
+  const stateClone = {};
+
+  Object.assign(stateClone, state);
+
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties':
+        {
+          const { extraData } = action;
+
+          Object.assign(stateClone, extraData);
+
+          results.push({ ...stateClone });
+        }
+        break;
+
+      case 'removeProperties':
+        {
+          const { keysToRemove } = action;
+
+          for (const key of keysToRemove) {
+            delete stateClone[key];
+          }
+
+          results.push({ ...stateClone });
+        }
+        break;
+
+      case 'clear':
+        for (const key of Object.keys(stateClone)) {
+          delete stateClone[key];
+        }
+
+        results.push({ ...stateClone });
+        break;
+    }
```

Comment:
The `switch` statement should always have a `default` case for error handling. This will ensure that the function behaves correctly even if an action with an unrecognized type is passed. [CODE STYLE]